requirements-git.txt

```diff
@@ -1,7 +1,7 @@
-git+https://github.com/firedrakeproject/ufl.git#egg=ufl
+git+https://github.com/firedrakeproject/ufl.git@dolci/merge_upstream_2
 git+https://github.com/firedrakeproject/fiat.git#egg=fiat
-git+https://github.com/FInAT/FInAT.git#egg=finat
-git+https://github.com/firedrakeproject/tsfc.git#egg=tsfc
+git+https://github.com/FInAT/FInAT.git@mscroggs/ufl-elements
+git+https://github.com/firedrakeproject/tsfc.git@mscroggs/newfl-legacy2
```

Comment:
Branch selection is meant to be done in `build.yml` as is done [here](https://github.com/firedrakeproject/firedrake/pull/3193/files). This could be why doc build is failing (@JDBetteridge  you're the expert there!) We can't merge whilst we have failing doc build. It would be handy to change this and retrigger tests before our meeting at 1600 UTC today where we will hopefully get this all merged

Reply:
I corrected how branch selection is done, docs still fail. Further suggestions welcome, I have no idea what I'm doing with the docs build

Reply:
This may not actually be a problem. We build the docs in containers so if we modify dependencies these changes are not known about by the container and then things won't resolve until we rebuild the containers.

Reply:
Don't worry Matt, I would expect our docs build to fail. It is currently set to build in a container with everything pre-installed at the default branch. We currently do not have a good (automatic) way to change the branch of dependencies when the docs build. If you can confirm that docs build locally (`make linkcheck && make html && make latexpdf` in the docs directory) that's more than what we insist on for most PRs!

Reply:
You'd need to run `firedrake-install --documentation-dependencies` first and move to the `docs` directory before running `make`. Also I've never managed to get `make latexpdf` working on my mac so I reckon as long as `make linkcheck && make html` works you're good to go.

Reply:
TBF we don't _need_ the LaTeX docs to build on mac, so long as they build on CI we can release PDF manuals.

Reply:
`make html` and `makelatexpdf` succeed locally.
`make linkcheck` complains that https://www.maths.ox.ac.uk/people/fabian.laakmann gives a 403 error. I guess this is unrelated to these changes

Reply:
Thanks, that link is working intermittently, so if it's a problem on CI later we'll fix it in another PR